src/trajectory.jl

```diff
@@ -557,31 +573,29 @@ Return the value with the largest absolute value.
 maxabs(a, b) = abs(a) > abs(b) ? a : b
 
 """
-    combine(treeleft::BinaryTree, treeright::BinaryTree)
-
+$(SIGNATURES)
 Merge a left tree `treeleft` and a right tree `treeright` under given Hamiltonian `h`,
 then draw a new candidate sample and update related statistics for the resulting tree.
 """
 function combine(treeleft::BinaryTree, treeright::BinaryTree)
     return BinaryTree(
         treeleft.zleft,
         treeright.zright,
-        combine(treeleft.c, treeright.c),
+        combine(treeleft.ts, treeright.ts),
         treeleft.sum_α + treeright.sum_α,
         treeleft.nα + treeright.nα,
         maxabs(treeleft.ΔH_max, treeright.ΔH_max),
     )
 end
 
 """
-    isterminated(h::Hamiltonian, t::BinaryTree{<:ClassicNoUTurn})
-
+$(SIGNATURES)
 Detect U turn for two phase points (`zleft` and `zright`) under given Hamiltonian `h`
 using the (original) no-U-turn cirterion.
 
 Ref: https://arxiv.org/abs/1111.4246, https://arxiv.org/abs/1701.02434
 """
-function isterminated(h::Hamiltonian, t::BinaryTree{<:ClassicNoUTurn})
+function isterminated(::ClassicNoUTurn, h::Hamiltonian, t::BinaryTree)
```

Comment:
Any reason for replace parametric `BinaryTree{AbstractTerminationCriterion}` with `isterminated(::ClassicNoUTurn, h::Hamiltonian, t::BinaryTree)`? Dispatching on parametric typed implementation is probably more natural than passing an additional type parameter..

Reply:
As I was not putting any type infomation about U-turn into turn stats.
By doing what you suggest on `TurnStatistic`, this problem can be resolve.